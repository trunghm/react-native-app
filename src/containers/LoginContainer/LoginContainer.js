'use strict'

import React from 'react'
import {connect} from 'react-redux'
import LoginComponent from './views/index'
import * as testActions from '../../actions/testActions'
import {bindActionCreators} from 'redux';

// Define which part of the state we're passing to this component
const mapStateToProps = (state) => ({})

// Define the actions this component may dispatch
const mapDispatchToProps = (dispatch) => {
  return {
    testActions: bindActionCreators(testActions, dispatch),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginComponent)
