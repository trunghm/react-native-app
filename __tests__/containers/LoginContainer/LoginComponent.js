/**
 * @format
 */

import 'react-native';
import React from 'react';
import {LoginComponent} from '../../../src/containers/LoginContainer/LoginContainer';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  LoginComponent component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <LoginComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <LoginComponent />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
