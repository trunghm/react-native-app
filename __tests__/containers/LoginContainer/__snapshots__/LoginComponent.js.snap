// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  LoginComponent component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  LoginComponent component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/themes/images/splash.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
          "marginTop": 163,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/themes/images/logo.png",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 20,
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#1496bb",
            "flex": 1,
            "justifyContent": "center",
            "maxHeight": 50,
            "minHeight": 50,
            "opacity": 1,
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          VIEW COMPONENTS
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 10,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 50,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#1496bb",
              "flex": 1,
              "justifyContent": "center",
              "maxHeight": 50,
              "minHeight": 50,
              "opacity": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
